acceptance/tests/utf8/utf8-recursive-copy.rb

```diff
@@ -0,0 +1,68 @@
+test_name "PUP-8735: UTF-8 characters are preserved after recursively copying directories" do
+  # Translation is not supported on these platforms:
+  confine :except, :platform => /^eos-/
+  confine :except, :platform => /^cisco/
+  confine :except, :platform => /^cumulus/
+  confine :except, :platform => /^solaris/
+
+  # for file_exists?
+  require 'puppet/acceptance/temp_file_utils'
+  extend Puppet::Acceptance::TempFileUtils
+
+  # for enable_locale_language
+  require 'puppet/acceptance/i18n_utils'
+  extend Puppet::Acceptance::I18nUtils
+
+  agents.each do |host|
+    filename = "Fișier"
+    content = <<-CONTENT
+閑けさや
+岩にしみいる
+蝉の声
+    CONTENT
+
+    workdir = host.tmpdir("tmp#{rand(999999).to_i}")
+    source_dir = "#{workdir}/Adresář"
+    target_dir = "#{workdir}/目录"
+
+    manifest = %Q|
+file { ["#{workdir}", "#{source_dir}"]:
+  ensure => directory,
+}
+
+file { "#{source_dir}/#{filename}":
+  ensure  => file,
+  content => "#{content}",
+}
+
+file { "#{source_dir}/#{filename}_Copy":
+  ensure => file,
+  source =>  "#{source_dir}/#{filename}",
+}
+
+file { "#{target_dir}":
+  ensure  => directory,
+  source  => "#{source_dir}",
+  recurse => remote,
+  replace => true,
+}|
+
+    step "Enable the non-UTF-8 en_US locale (and skip this test if we can't)" do
+      host_language = enable_locale_language(host, 'en_US', prefer_utf8: false)
+      if host_language.nil?
+        skip_test("Host #{host} is missing the #{host_language} locale. Skipping this test.")
```

Comment:
Oops, this message will have "nil" as the host language. I'll fix as soon as Jenkins PR tests run through.